base/math.jl

```diff
@@ -146,6 +146,10 @@ sqrt(x::Float32) = box(Float32,sqrt_llvm(unbox(Float32,x)))
 sqrt(x::Real) = sqrt(float(x))
 @vectorize_1arg Number sqrt
 
+# 1-arg hypot
+hypot(x::Real) = abs(promote(float(x))...)
```

Comment:
You would need to return `promote(float(x))` also if `x` is nan, for type stability. `abs` is quite cheap, probably cheaper than `isnan` actually, so this doesn't look like an improvement.